docs/src/main/asciidoc/_high-availability/cloud/Kubernetes.adoc

```diff
@@ -0,0 +1,230 @@
+REVISION: 1
+TEST SUITE: None
+NOTES:
+HAProxy Ingress has been installed!
+
+HAProxy is exposed as a `LoadBalancer` type service.
+It may take a few minutes for the LoadBalancer IP to be available.
+You can watch the status by running:
+
+    kubectl --namespace ingress-controller get services haproxy-ingress -o wide -w
+----
+
+Once the controller is available in the Kubernetes cluster, we can configure an Ingress object.
+
+=== Configuring an Ingress
+
+The following Ingress configuration can be used for optimal handling of session affinity.
+Note that the name of the ingress, hosts, and the service name needs to be updated to actual values.
+
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: example-ingress
+  namespace: default
+  annotations:
+    haproxy-ingress.github.io/affinity: cookie # <1>
+    haproxy-ingress.github.io/backend-server-naming: pod # <2>
+    haproxy-ingress.github.io/session-cookie-dynamic: "false" # <3>
+    haproxy-ingress.github.io/session-cookie-keywords: preserve # <4>
+    haproxy-ingress.github.io/session-cookie-name: INGRESSCOOKIE # <5>
+    haproxy-ingress.github.io/session-cookie-preserve: "true" # <6>
+    haproxy-ingress.github.io/session-cookie-strategy: insert # <7>
+    haproxy-ingress.github.io/session-cookie-value-strategy: server-name # <8>
+spec:
+  ingressClassName: haproxy
+  tls:
+    - hosts:
+        - example.com
+  rules:
+    - host: example.com
+      http:
+        paths:
+          - path: /
+            pathType: Prefix
+            backend:
+              service:
+                name: example-service
+                port:
+                  number: 8080
+----
+<1> Use a cookie to store the affinity.
+<2> Use the pod's unique name as the backend server identity. This allows WildFly to update the existing affinity.
+<3> Instructs the proxy to use a predictable backend server name.
+<4> Configures preserving cookies as additional options for handling cookies.
+<5> The name of the cookie that contains the affinity information.
+<6> Allow backend servers to use a Set-Cookie HTTP header to emit their own affinity cookie value, meaning the backend servers have knowledge of which cookie value should route to which server.
+<7> Instructs the proxy to not modify/rewrite existing affinity cookie, only insert if a cookie is not set (e.g. if the backed server does not support storing affinity in a separate cookie)
+<8> Instructs the proxy what names to use as backed service names by respecting the `backend-server-naming` setting.
+
+== Verifying Session Affinity
+
+Verifying session affinity has proven to be notoriously challenging for users.
+This is primarily because of the fact, that even if the affinity handling is not set up correctly,
+the observer will not observe any data loss in a distributed HA application.
+This is simply because if the request for a given HTTP session is routed to a cluster node that does not have the session data locally,
```

Comment:
"This is simply because, if the request for a given HTTP session is routed to a ~~cluster node~~ _pod_ that does not ~~have~~ contain the _requested_ session data locally,"

Reply:
Fixed.